cpp/src/neighbors/mg/snmg.cuh

```diff
@@ -107,8 +121,18 @@ void build(const raft::resources& clique,
     RAFT_LOG_DEBUG("SHARDED BUILD: %d*%drows", index.num_ranks_, n_rows_per_shard);
 
     index.ann_interfaces_.resize(index.num_ranks_);
+
+    // Enable nested parallelism: build uses OpenMP internally
+    int saved_omp_threads = omp_get_max_threads();
+    int threads_per_rank  = saved_omp_threads / index.num_ranks_;
```

Comment:
This seems risky. If saved_omp_threads is less than index.num_threads_, threads_per_rank will be set to zero. Instead we can do this:
`threads_per_rank = max(1, saved_omp_threads / index.num_ranks_);`